packages/core/src/deployment.ts

```diff
@@ -73,11 +74,15 @@ export async function waitAndValidateDeployment(provider: EthereumProvider, depl
   }
 
   debug('succeeded verifying deployment', txHash);
-  if (await hasCode(provider, address)) {
-    return;
+  while (!(await hasCode(provider, address))) {
```

Comment:
My only concern with this PR is that the original reason we check `hasCode` is that if the transaction was mined but `hasCode` returns empty it was because the transaction failed. This PR seems to add a 1 minute delay to detect a failed transaction. Am I understanding that correctly?

Reply:
Makes sense. How about I add a check for the tx status in the receipt, and only await this 1 minute if the tx was flagged as successful?